src/commissaire_http/__init__.py

```diff
@@ -14,17 +14,96 @@
+    the configuration file.
+
+    The 'parser' argument should be a fresh argparse.ArgumentParser
+    instance with a suitable description, epilog, etc.  This method
+    will add arguments to it.
+
+    :param parser: An argument parser instance
+    :type parser: argparse.ArgumentParser
+    :returns: The parsed arguments in the form of a Namespace
+    :rtype: argparse.Namespace
+    """
+    # Do not use required=True because it would preclude such
+    # arguments from being specified in a configuration file.
+    parser.add_argument(
+        '--config-file', '-c', type=str,
+        help='Full path to a JSON configuration file '
+             '(command-line arguments override)')
+    parser.add_argument(
+        '--no-config-file', action='store_true',
+        help='Disregard default configuration file, if it exists')
+    parser.add_argument(
+        '--listen-interface', '-i', type=str, default='0.0.0.0',
+        help='Interface to listen on')
+    parser.add_argument(
+        '--listen-port', '-p', type=int, default=8000,
+        help='Port to listen on')
+    parser.add_argument(
+        '--tls-keyfile', type=str,
+        help='Full path to the TLS keyfile for the commissaire server')
+    parser.add_argument(
+        '--tls-certfile', type=str,
+        help='Full path to the TLS certfile for the commissaire server')
+    parser.add_argument(
+        '--tls-clientverifyfile', type=str,
+        help='Full path to the TLS file containing the certificate '
+             'authorities that client certificates should be verified against')
+    parser.add_argument(
+        '--authentication-plugin', type=str,
+        default='commissaire.authentication.httpbasicauth',
+        metavar='MODULE_NAME',
+        help=('Authentication Plugin module. '
+              'EX: commissaire.authentication.httpbasicauth'))
+    parser.add_argument(
+        '--authentication-plugin-kwargs', type=str, default={},
+        metavar='KEYWORD_ARGS',
+        help='Authentication Plugin configuration (key=value,...)')
+    parser.add_argument(
+        '--store-handlers', type=str, default=[],
+        action='append', metavar='JSON_OBJECT',
+        help='Store Handler configuration in JSON format, '
+             'can be specified multiple times')
+
+    parser.add_argument(
+        '--bus-exchange', type=str, default='commissaire',
+        help='Message bus exchange name.')
+    parser.add_argument(
+        '--bus-uri', type=str, metavar='BUS_URI',
+        help=(
+            'Message bus connection URI. See:'
+            'http://kombu.readthedocs.io/en/latest/userguide/connections.html')
+    )
+
+    # We have to parse the command-line arguments twice.  Once to extract
+    # the --config-file option, and again with the config file content as
+    # a baseline.
+    args = parser.parse_args()
+    if not args.no_config_file:
+        namespace_kwargs = read_config_file(args.config_file)
```

Comment:
I changed `read_config_file` to return a plain dictionary now instead of an `argparse.Namespace`.

I think that's more appropriate for `commissaire-service` because I don't see the backend services having much of a command-line interface; they just read the JSON config.

The conversion here would be:
```
json_object = read_config_file(args.config_file)

# Normalize member names by converting hypens to underscores.
json_object = {k.replace('-', '_'): v for k, v in json_object.items()}

namespace_kwargs = argparse.Namespace(**json_object)
```